### PR TITLE
Rate limit uphold

### LINKED ISSUE #476: Avoid hitting uphold API limits
In the past we've caused DOS'd ourselves by hitting our API limits on uphold, e.g. due to balance server issue having all balance queries poll through to uphold
a circuit breaker pattern there with max in-flight requests could be useful there

### PR DESCRIPTION
**Pull request by maikelmclauflin**
fixes #476

### CODE REVIEW

File: bat-utils/lib/extras-utils.js [hunk 1]
```diff
@@ -2,13 +2,38 @@ const getPublisherProps = require('bat-publisher').getPublisherProps
...
+const ledgerLimiter = new Bottleneck({
+  reservoir: 50, // initial value
+  reservoirRefreshAmount: 25,
+  reservoirRefreshInterval: 10 * 1000, // must be divisible by 250
+  // also use maxConcurrent and/or minTime for safety
+  maxConcurrent: 32,
+  minTime: 0.1
+})
+
+const upholdLimiter = new Bottleneck({
```
Comment (evq) [inline bat-utils/lib/extras-utils.js]: these limits are per instance - correct?

Comment (maikelmclauflin) [reply bat-utils/lib/extras-utils.js]: yes. they would end up being per instance. i believe i can set it up to be shared across a redis store as well if that is preferrable

File: .travis.yml [hunk 2]
```diff
@@ -6,6 +6,7 @@ branches:
   - master
 env:
   global:
+    - "SERVICE=ledger"
```
Comment (evq) [inline .travis.yml]: hey why didn't we need this before?

Comment (maikelmclauflin) [reply .travis.yml]: may be a relic from when i was debugging something. shall investigate

Comment (maikelmclauflin) [reply .travis.yml]: this was so that utils tests that require configuration do not fail

**Pull request closed without merging**